### PR TITLE
CE-94 revisited: Adding PID class with tests

### PR DESCRIPTION
**Pull request by ialarmedalien**
PermanentID class:
id: mandatory, must match regex
description: optional, free text
relationshipType: optional, currently free text but may be restricted to a controlled vocab in the future

### CODE REVIEW

File: src/us/kbase/workspace/database/provenance/PermanentID.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	 * @return the description, if present.
+	 */
+	public Optional<String> getDescription() {
+		return Optional.ofNullable(description);
+	}
+
+	/**
+	 * Gets the relationship between the ID and some other entity. For
+	 * example, when a {@link PermanentID] class is used to represent
+	 * objects in a {@link Resource}'s 'related_identifiers' field, this
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/provenance/PermanentID.java:45]: I presume this'll be added in a later PR

File: src/us/kbase/workspace/database/provenance/PermanentID.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	public Optional<String> getRelationshipType() {
+		return Optional.ofNullable(relationshipType);
+	}
+
+	/**
+	 * Gets a builder for an {@link PermanentID}.
+	 *
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(final String id) {
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/provenance/PermanentID.java:63]: Needs `@param` entry for an id describing the format

**Review by MrCreosote — CHANGES_REQUESTED** [review]
One minor doc update. After that, approval is tentative pending
* tests pass
* coverage is 100% as expected
* the missing coverage in the Common regex matching functions is now covered as discussed

File: src/us/kbase/workspace/database/provenance/PermanentID.java (hunk 1 above)
Comment (ialarmedalien) [reply src/us/kbase/workspace/database/provenance/PermanentID.java:45]: Yup, correct.

File: src/us/kbase/workspace/database/provenance/PermanentID.java (hunk 2 above)
Comment (ialarmedalien) [reply src/us/kbase/workspace/database/provenance/PermanentID.java:64]: Done.

**Pull request merged**